dbms/src/Common/MPMCQueue.h

```diff
@@ -289,26 +302,34 @@ class MPMCQueue
                 return read_pos < write_pos || !isNormal();
             };
             if (!wait(lock, reader_head, node, pred, deadline))
-                return Result::TIMEOUT;
+                is_timeout = true;
         }
         if (!isCancelled() && read_pos < write_pos)
         {
             auto & obj = getObj(read_pos);
             res = std::move(obj);
             destruct(obj);
+            current_auxiliary_memory_usage -= get_auxiliary_memory_usage(res);
```

Comment:
It's a little wasteful to call `get_auxiliary_memory_usage` twice for one object.
And also it requires that these two return values of `get_auxiliary_memory_usage` must be the same.

Reply:
OK, save `auxiliary_memory` in another vector.